pkg/services/profiles/add.go

```diff
@@ -0,0 +1,175 @@
+	if err != nil {
+		return err
+	}
+
+	// TODO: refactor this into 1 func that returns repoRef
+	repoExists, err := gitProvider.RepositoryExists(ctx, configRepoUrl)
+	if err != nil {
+		return err
+	} else if !repoExists {
+		return fmt.Errorf("repository '%v' could not be found", configRepoUrl.String())
+	}
+	defaultBranch, err := gitProvider.GetDefaultBranch(ctx, configRepoUrl)
+	if err != nil {
+		return err
+	}
+
+	// TODO: We need a ticket for when the user auto-completes the chart name, helm repository name, and helm repository namespace,
+	// so that a user doesn't only rely on hitting the Kube API if they don't have access to the cluster.
+	availableProfile, err := s.GetAvailableProfile(ctx, GetOptions{
+		Name:      opts.Name,
+		Version:   opts.Version,
+		Cluster:   opts.Cluster,
+		Namespace: opts.Namespace,
+	})
+	if err != nil {
+		return err
+	}
+	if availableProfile.HelmRepository == nil || availableProfile.HelmRepository.Name == "" || availableProfile.HelmRepository.Namespace == "" {
+		return fmt.Errorf("failed to discover HelmRepository's name and namespace")
+	}
+
+	newRelease := helm.MakeHelmRelease(availableProfile, opts.Cluster, opts.Namespace)
+
+	files, err := gitProvider.GetRepoFiles(ctx, configRepoUrl, git.GetSystemPath(opts.Cluster), defaultBranch)
+	if err != nil {
+		return fmt.Errorf("failed to get files in '%s' for config repository '%s': %s", git.GetSystemPath(opts.Cluster), configRepoUrl.String(), err)
+	}
+
+	file, err := MakeManifestFile(files, newRelease, git.GetProfilesPath(opts.Cluster))
+	if err != nil {
+		return err
+	}
+
+	pr, err := gitProvider.CreatePullRequest(ctx, configRepoUrl, gitproviders.PullRequestInfo{
+		Title:         fmt.Sprintf("GitOps add %s", opts.Name),
+		Description:   fmt.Sprintf("Add manifest for %s profile", opts.Name),
+		CommitMessage: AddCommitMessage,
+		TargetBranch:  defaultBranch,
+		NewBranch:     automation.GetRandomName("wego-"),
+		Files:         file,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create the pull request: %s", err)
+	}
+	s.Logger.Actionf("Pull Request created: %s", pr.Get().WebURL)
+
+	if opts.AutoMerge {
+		s.Logger.Actionf("auto-merge=true; merging PR number %v", pr.Get().Number)
+		if err := gitProvider.MergePullRequest(ctx, configRepoUrl, pr.Get().Number, gitprovider.MergeMethodMerge, AddCommitMessage); err != nil {
+			return err
+		}
+	}
+
+	s.printAddSummary(opts)
+	return nil
+}
+
+func (s *ProfilesSvc) printAddSummary(opts AddOptions) {
+	s.Logger.Println("Adding profile:\n")
+	s.Logger.Println("Name: %s", opts.Name)
+	s.Logger.Println("Version: %s", opts.Version)
+	s.Logger.Println("Cluster: %s", opts.Cluster)
+	s.Logger.Println("Namespace: %s", opts.Namespace)
+	s.Logger.Println("")
+}
+
+func MakeManifestFile(files []*gitprovider.CommitFile, newRelease *v2beta1.HelmRelease, path string) ([]gitprovider.CommitFile, error) {
+	var content string
+	for _, f := range files {
+		if *f.Path == path {
+			if *f.Content == "" {
```

Comment:
to avoid any problems, use `len(f.Content) == 0` here, because that returns 0 even if Content is nil. Which I see that is a required field, but still. :) No need to paint the devil on the wall.